BUG: fix elp montage rotation

jona-sassenhagen opened the pull request:
With .elp channel locations, without this patch:

![unknown-75](https://cloud.githubusercontent.com/assets/4321826/20154431/89ee194a-a6c7-11e6-9446-207cc6dcef06.png)

with this patch:

![unknown-8](https://cloud.githubusercontent.com/assets/4321826/20154414/7cba3538-a6c7-11e6-906d-80f73086e8af.png)

Possibly related to #3613 and/or #3690 ?

@Eric89GXL can you check? I'm really not sure.

jona-sassenhagen commented:
```
======================================================================
FAIL: Test topo to sphere conversion.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\conda\envs\test\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\projects\mne-python\mne\tests\test_transforms.py", line 222, in test_topo_to_sph
    assert_allclose(pos, new[ii], atol=1e-7)
  File "C:\conda\envs\test\lib\site-packages\numpy\testing\utils.py", line 1297, in assert_allclose
    verbose=verbose, header=header)
  File "C:\conda\envs\test\lib\site-packages\numpy\testing\utils.py", line 665, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=1e-07
(mismatch 100.0%)
 x: array([ 0.353553,  0.612372,  0.707107])
 y: array([-0.612372, -0.353553,  0.707107])
>>  raise AssertionError('\nNot equal to tolerance rtol=1e-07, atol=1e-07\n\n(mismatch 100.0%)\n x: array([ 0.353553,  0.612372,  0.707107])\n y: array([-0.612372, -0.353553,  0.707107])')
```

Not sure what is going on.
@wmvanvliet maybe you have an idea?

teonbrooks commented:
I agree that the fix makes the plots look correct but the original equations seem correct. do you know what's going on here? is the `.elp` coordinate system like left-handed?

Eric89GXL commented:
@jona-sassenhagen can you check with `git bisect` where it went wrong? I'm guessing it was one of my PRs... it would be good to beef up the tests more to make sure things get translated properly. If you don't have time let me know and I can look into it since I probably broke it.

agramfort commented:
FYI I've seen this type of rotations using eeglab .set files

cc @jmontoyam

jasmainak commented:
yeah, the fix doesn't look correct to me. Where did you get this `.elp` file from? Possible to add a test?

mmagnuski commented:
I've seen this types of rotations when converting eeglab files to fieldtrip (`eeglab2fieltrip` IIRC, but that was some time ago - that might have been fixed there already. eeglab and fieldtrip  topoplots were giving different head rotations at that time [at least for EGI channel locs])

agramfort commented:
I bet it's a bug in the file conversion but not in MNE-python no?

jona-sassenhagen commented:
The error occurs with multiple Brainvision files and .elp files. These specific files gave correct results with previous versions of MNE.

I will try to find what version is to blame.

jona-sassenhagen commented:
I can't really pinpoint the error because of the error introduced in #3613 and fixed in #3690, because the error specifically affects .elp reading. It is possible I introduced this specific error in my fix in #3690.

The last commit before #3613 works (2fa35d453670d759df3ff296c1f5c012f3f8a785).
Then #3613 makes .elps unreadable.
From #3690 on, .elps can be read, but they are rotated (when I did #3690, I only checked central topos that didn't change much under rotation :().

This also indicates IMO that the test modified in #3690 may be insufficient ... as it relies too much on shapes rather than values of the returned arrays.

@Eric89GXL can you indeed take a look? #3613 is very long and I'm really bad with transforms ...
